In [37]:
# Author: GreHiDeL

In [39]:
import os
import pandas as pd
from binance import Client
api_key = os.environ.get('BINANCE_TEST_API')
api_secret = os.environ.get('BINANCE_TEST_SECRET')
client=client = Client(api_key, api_secret)

In [40]:
data=client.get_historical_klines("BNBBUSD", Client.KLINE_INTERVAL_1DAY, "4 day ago UTC+3")  #https://python-binance.readthedocs.io/en/latest/market_data.html?highlight=kline%20limit#id6
frame=pd.DataFrame(data)
frame=frame.iloc[:,:6] # All rows and column upto 6
# Naming the columns
frame.columns=['Time','Open','High','Low','Close','Volume']
# Name the rows
frame=frame.set_index('Time') # Note this is in ms
# Convert this Time to readable form since this is time from 1970s
frame.index=pd.to_datetime(frame.index,unit='ms')
frame=frame.astype(float) # Convert values from strings to float
frame

Open   High    Low  Close      Volume
Time                                              
2022-04-29  407.2  410.7  389.7  392.8  233647.232
2022-04-30  392.9  399.8  376.4  377.2  154503.970
2022-05-01  377.2  391.3  375.6  389.9  165113.527
2022-05-02  390.0  392.6  386.9  389.4   37560.666

Conclusion: On 29th April 202 the Price closed at 392.8 while the settlement price was 4**